extensions/ipynb/src/deserializers.ts

```diff
@@ -61,8 +69,19 @@ function sortOutputItemsBasedOnDisplayOrder(outputItems: NotebookCellOutputItem[
 			}
 			return compareWith.startsWith(value);
 		};
-		const indexOfMimeTypeA = orderOfMimeTypes.findIndex(mime => isMimeTypeMatch(outputItemA.mime, mime));
-		const indexOfMimeTypeB = orderOfMimeTypes.findIndex(mime => isMimeTypeMatch(outputItemB.mime, mime));
+		let indexOfMimeTypeA = orderOfMimeTypes.findIndex((mime) => isMimeTypeMatch(mime, outputItemA.mime));
```

Comment:
Added something that was fixed in Jupyter extension (some how this was missing here).
I've added the changes/fixes here along with the corresponding tests.